lib/Yam/Agama/Pom/RebootPage.pm

```diff
@@ -27,7 +28,12 @@ sub expect_is_shown {
 
 sub reboot {
     my ($self) = @_;
-    assert_and_click($self->{tag_reboot_button});
+    if (is_s390x()) {
+        enter_cmd 'reboot';
+    }
+    else {
+        assert_and_click($self->{tag_reboot_button});
+    }
```

Comment:
so, in the test layer we should avoid to create conditions (if/else) but in the POM layer we must not create any conditions, because otherwise it will defeat its purpose, this design is thought to move the conditions away among other things.

For s390x we need a new class where `expect_is_shown`  has a different implementation (for example checking the logs, we can do that in another ticket for sure) and also `reboot` has a different one, doing it by the command line, it is kind of like class RebootTextmodePage for example and you put the condition in `lib/Distribution/Opensuse/AgamaDevel.pm`

Reply:
5df52e594bdbf216d60854b5e3a2e2405a4b14f3